xarray/tests/test_accessors.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+try:
+    import cPickle as pickle
+except ImportError:
+    import pickle
+
+import xarray as xr
+import numpy as np
+import pandas as pd
+
+from . import TestCase
+
+
+class TestDatetimeAccessor(TestCase):
+    def setUp(self):
+        nt = 10000
+        data = np.random.rand(10, 10, nt)
+        lons = np.linspace(0, 11, 10)
+        lats = np.linspace(0, 20, 10)
+        self.times = pd.date_range(start="2000/01/01", freq='H', periods=nt)
+
+        self.data = xr.DataArray(data, coords=[lons, lats, self.times],
+                                 dims=['lon', 'lat', 'time'], name='data')
+
+    def test_field_access(self):
+        years = self.times.year
+        months = self.times.month
+        days = self.times.day
+        hours = self.times.hour
+
+        self.assertArrayEqual(years, self.data.time.dt.year)
```

Comment:
Use `self.assertDataArrayEqual` to check that fields provide data arrays with the appropriate metadata.